docs/source/developers/python.rst

```diff
@@ -586,6 +586,86 @@ Caveats
 
 The Plasma component is not supported on Windows.
 
+Relevant components and environment variables
+=============================================
+
+List of relevant Arrow CMake flags and corresponding environment variables
+to be used when building PyArrow are:
+
+.. list-table::
+   :widths: 30 30
+   :header-rows: 1
+
+   * - CMAKE_BUILD_TYPE
+     - PYARROW_BUILD_TYPE (release, dbug or relwithdebinfo)
+   * - ARROW_GCS
+     - PYARROW_WITH_GCS
+   * - ARROW_S3
+     - PYARROW_WITH_S3
+   * - ARROW_HDFS
+     - PYARROW_WITH_HDFS
+   * - ARROW_CUDA
+     - PYARROW_WITH_CUDA
+   * - ARROW_SUBSTRAIT
+     - PYARROW_WITH_SUBSTRAIT
+   * - ARROW_FLIGHT
+     - PYARROW_WITH_FLIGHT
+   * - ARROW_DATASET
+     - PYARROW_WITH_DATASET
+   * - ARROW_PARQUET
+     - PYARROW_WITH_PARQUET
+   * - PARQUET_REQUIRE_ENCRYPTION
+     - PYARROW_WITH_PARQUET_ENCRYPTION
+   * - ARROW_PLASMA
+     - PYARROW_WITH_PLASMA
+   * - ARROW_TENSORFLOW
+     - PYARROW_WITH_TENSORFLOW
+   * - ARROW_ORC
+     - PYARROW_WITH_ORC
+   * - ARROW_GANDIVA
+     - PYARROW_WITH_GANDIVA
+
+List of relevant environment variables that can also be used to build
+PyArrow are:
+
+.. list-table::
+   :widths: 30 30
+   :header-rows: 1
+
+   * - PyArrow environment variable
+     - Description
+   * - PYARROW_CMAKE_GENERATOR
+     - Example: 'Visual Studio 15 2017 Win64'
+   * - PYARROW_CMAKE_OPTIONS
+     - Extra CMake and Arrow options (ex. ``"-DARROW_SIMD_LEVEL"``,
+       ``"-DCMAKE_OSX_ARCHITECTURES"``)
+   * - PYARROW_BOOST_NAMESPACE
+     - Name of install folder for boost (if different from 'boost')
+   * - PYARROW_CXXFLAGS
+     - Extra cxx flags
+   * - PYARROW_WITH_STATIC_PARQUET
+     - Rely on parquet shared libraries where relevant, default OFF
+   * - PYARROW_WITH_STATIC_BOOST
+     - Rely on Boost shared libraries on linking static parquet, default OFF
+   * - PYARROW_GENERATE_COVERAGE
+     - Setting ``Xlinetrace`` flag to true for the Cython compiler
+   * - PYARROW_BUNDLE_ARROW_CPP
+     - Bundle the Arrow C++ libraries, default OFF
+   * - PYARROW_BUNDLE_CYTHON_CPP
+     - Bundle the C++ files generated by Cython, default OFF
+   * - PYARROW_BUNDLE_BOOST
+     - Bundle the Boost libraries when we bundle Arrow C++, default OFF
```

Comment:
I should have removed this in #14925.

This is not used now because Apache Arrow C++'s library part doesn't depend on Boost libraries. (Tests may depend on Boost libraries. Apache Arrow C++'s library part may depend on Boost but it just uses header-only part of Boost.)

Reply:
Thanks for the explanation!
I can remove the variable from setup.py in this PR if you agree it fits.